src/ducc0/infra/threading.h

```diff
@@ -53,21 +53,106 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef DUCC0_THREADING_H
 #define DUCC0_THREADING_H
 
+// Low level threading support can be influenced by the following macros:
+// - DUCC0_NO_LOWLEVEL_THREADING: if defined, multithreading is disabled
+//   and all parallel regions will be executed sequentially
+//   on the invoking thread.
+// - DUCC0_CUSTOM_LOWLEVEL_THREADING: if defined, external definitions of
+//   Mutex, UniqueLock, LockGuard, CondVar, thread_pool, set_active_pool(),
+//   and get active_pool() must be supplied,
+//   and the code will use those.
+// Both macros must not be defined at the same time.
+// If neither macro is defined, standard ducc0 multihreading will be active.
+
+#if (defined(DUCC0_NO_LOWLEVEL_THREADING) && defined(DUCC0_CUSTOM_LOWLEVEL_THREADING))
+static_assert(false, "DUCC0_NO_LOWLEVEL_THREADING and DUCC0_CUSTOMLOWLEVEL_THREADING must not be both defined");
+#endif
+
+#if defined(DUCC0_STDCXX_LOWLEVEL_THREADING)
+static_assert(false, "DUCC0_STDCXX_LOWLEVEL_THREADING must not be defined externally");
+#endif
+
+#if ((!defined(DUCC0_NO_LOWLEVEL_THREADING)) && (!defined(DUCC0_CUSTOM_LOWLEVEL_THREADING)))
+#define DUCC0_STDCXX_LOWLEVEL_THREADING
+#endif
+
 #include <cstddef>
 #include <functional>
-#ifndef DUCC0_NO_THREADING
+#include <optional>
+
+// threading-specific headers
+#ifdef DUCC0_STDCXX_LOWLEVEL_THREADING
 #include <mutex>
 #include <condition_variable>
 #endif
-#include <optional>
+
+#ifdef DUCC0_NO_LOWLEVEL_THREADING
+// no headers needed
+#endif
+
+#ifdef DUCC0_CUSTOM_LOWLEVEL_THREADING
+// include headers necessary for custom threading support here.
```

Comment:
I suggest to explicit put a filename here, like
```
#include "ducc0_custom_lowlevel_threading.h"
```
That way the user can provide a header and ensure it is on the set of include paths so that it gets picked up.  Otherwise, the user has no choice put to patch/modify your actual library source.

Reply:
OK, so far I was assuming that some amount of source-changing would be necessary, but I see what you mean.